app/src/parser_special.c

```diff
@@ -134,37 +152,118 @@ parser_error_t parser_getItem_NativeTransfer(ExecutableDeployItem item, parser_c
                                              outVal, outValLen,
                                              pageIdx, pageCount);
         }
-
-        return parser_no_data;
     }
+
+    return parser_no_data;
 }
 
-parser_error_t parseNativeTransfer(parser_context_t *ctx, ExecutableDeployItem *item, uint32_t num_items) {
-    PARSER_ASSERT_OR_ERROR(3 <= num_items && num_items <= 4, parser_unexpected_number_items);
+
+parser_error_t checkNativeTransferArgs(parser_context_t *ctx, __Z_UNUSED ExecutableDeployItem *item, uint32_t num_items, uint32_t *fitems) {
+
     uint8_t type = 0;
     uint8_t internal_type = 0;
-    CHECK_RUNTIME_ARGTYPE(ctx, num_items, "amount", type == 8);
-    CHECK_RUNTIME_ARGTYPE(ctx, num_items, "id", type == 13 && internal_type == 5);
-    CHECK_RUNTIME_ARGTYPE(ctx, num_items, "target", type == 11 || type == 12 || type == 15 || type == 22);
+    *fitems = 0;
+    parser_error_t err = searchRuntimeArgs("amount", &type, &internal_type, num_items, ctx);
+
+    if (err == parser_ok && (type == 8 || type == 4 || type == 5) ) {
+        *fitems += 1;
+    } else if (err != parser_runtimearg_notfound && err != parser_unexpected_type)
```

Comment:
I hope we can combine "!=  parser_runtimearg_notfound and  != parser_unexpected_type into a unique error-code?